src/track/beats.h

```diff
@@ -74,8 +74,32 @@ class Beats : private std::enable_shared_from_this<Beats> {
             updateValue();
         }
 
+        /// Returns the frame duration between this beat and the next beat.
         mixxx::audio::FrameDiff_t beatLengthFrames() const;
 
+        /// Returns the number of beats per bar.
+        ///
+        /// Note: Currently always returns 4, as time signatures other than 4/4
+        /// are not supported yet.
+        int beatsPerBar() const {
+            return 4;
+        }
+
+        /// Returns the current index of the beat in the bar.
+        int beatIndexInBar() const {
+            return m_beatOffset % beatsPerBar();
+        }
+
+        /// Returns true if this beat is at a marker position.
+        bool isMarker() const {
+            return m_beatOffset == 0;
+        }
+
+        /// Returns true if this beat is a downbeat.
+        bool isDownbeat() const {
+            return beatIndexInBar() == 0;
```

Comment:
I really like to see the SNMR notation natively build in. 
https://georgboenn.blogspot.com/2018/04/research-on-rhythm-and-meter.html

Nothing for this PR, but I am afraid this sets a foundation that we can't integrate it easily later. 

Do you have interest to discuss the options on Zulip? 

For my feeling it is the right time to integrate it. 

An additional smaller invisible grid than a beat, something like a 1/64 note or such and than a 64 bit field in to mark the beats on this grid. A 4/4 measure becomes than a SNMR of 1000000000000000100000000000000010000000000000001000000000000000
0x8000800080008000

This introduces only a little complexity for now but we are open for complex looping and transitions later.

Reply:
I don't think this is too hard to implement later.

And to be honest, I'd rather have an incremental development approach, so that we have an implementation that works for the 4/4 case now.

Otherwise I fear that we are overengineering it, because IMHO there are too many open questions atm. How would sync between multiple time signatues work (align downbeats? align beats?). This becomes even more complicated with rhythmic phrases that this format allows.

I also don't think the example you gave is correct, at least according to the blog post you linked, a 4/4 should be `10101010`. `100` is already a dotted quarter note, and I think the other zeroes are probably 8th pauses or something.

For those reasons, I think it would be premature to integrate it at this point.

Reply:
I also think this SNMR concept is over-engineered and will create a large amount of complexity for not a lot of gain.  It is not the responsibility of Mixxx to become a savant at complex polymeter beatmatching, and any choices we make are likely to be incorrect anyway.  (How the heck would Sync Lock work??).  Any DJ working with those kind of tracks would be musically talented enough that they are doing their own syncing anyway.  Let's go with this incremental approach and iterate from there.  

(Just as an example, even a simple concept like half/double BPM matching has been a humongous source of pain and bugs in Sync Lock. I can't imagine what trying to maintain a more complex sub-beat-matching system would be like -- I daresay it would be functionally impossible to get right)

Reply:
Of cause this is nothing for this PR. I will address your concerns on Zulip. 
https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/SNMR.20notation